include/oneapi/dpl/internal/dynamic_selection_impl/auto_tune_policy.h

```diff
@@ -217,6 +217,10 @@ class auto_tune_policy
         static_assert(sizeof...(KeyArgs) == sizeof...(Args));
         if (state_)
         {
+            if (state_->resources_with_index_.empty())
+            {
+                initialize();
```

Comment:
This will be fine for backends like sycl that can auto-initialize resources. However, for other backends that resources are manually initialized, this will fail. A message indicating that "no devices were found" or some combination of the two might be better.

Reply:
Fixed

Reply:
How would we get into a situation where state_ is not null and resources_with_index is empty, but calling initialize might still succeed?  I'm not sure what scenario this PR is trying to address?

Reply:
The case would be that initialize from the backend (not be a sycl_backend, something in the future)  would return 0 resources.